corehq/apps/app_manager/static/app_manager/js/case-config-ui-advanced.js

```diff
@@ -950,6 +952,7 @@ var AdvancedCase = (function () {
 
             self.removeProperty = function (property) {
                 self.case_properties.remove(property);
+                self.config.saveButton.fire('change');
```

Comment:
Ha, I thought I had tried this and it didn't work for me. I am sure you have tested it.